### ISSUE TITLE
Allow message content to change when handling actions

### ISSUE THREAD

**Issue by theothertomelliott** [theothertomelliott/spanner#17]
Currently the following code looks like it should update the message (as with views) but does not:

```
if msg := ev.ReceiveMessage(); msg != nil && msg.Text() == "hello" {
	reply := ev.SendMessage(msg.Channel().ID())
	letter := reply.Select("Pick a letter", spanner.Options("a", "b", "c"))
	if letter != "" {
		reply.PlainText(fmt.Sprintf("You chose: %v", letter))
	}
}
```

**Comment by theothertomelliott** [comment]
This works in the unit tests, but not the application itself.

**Comment by theothertomelliott** [comment]
The above issue was fixed in PR #19, but there is an error if you try to update a parent message after a child message has been send.
